datalad/distribution/create.py

```diff
@@ -264,29 +280,31 @@ def __call__(
             # comes around
             gitattr.write('** annex.largefiles=nothing\n')
 
-        # save everthing
-        tbds.add('.datalad', to_git=True, save=False)
-
-        if save:
-            save_dataset(
-                tbds,
-                paths=['.datalad'],
-                message='[DATALAD] new dataset')
-
-            # the next only makes sense if we saved the created dataset,
-            # otherwise we have no committed state to be registered
-            # in the parent
-            if dataset is not None and dataset.path != tbds.path:
-                # we created a dataset in another dataset
-                # -> make submodule
-                dataset.add(tbds.path, save=save, ds2super=True)
-
-        return tbds
+        # save everything
+        tbds.add('.datalad', to_git=True, save=save,
+                 message='[DATALAD] new dataset')
+
+        # the next only makes sense if we saved the created dataset,
+        # otherwise we have no committed state to be registered
+        # in the parent
+        if save and dataset is not None and dataset.path != tbds.path:
+            # we created a dataset in another dataset
+            # -> make submodule
+            dataset.add(tbds.path, save=save, ds2super=True)
+
+        yield get_status_dict(
+            action='create',
+            ds=tbds,
+            logger=lgr,
+            refds=refds_path,
+            status='ok')
 
     @staticmethod
-    def result_renderer_cmdline(res, args):
+    def custom_result_renderer(res, **kwargs):
         from datalad.ui import ui
-        if res is None:
+        if res.get('action', None) == 'create' and \
+               res.get('status', None) == 'ok' and \
+               res.get('type', None) == 'dataset':
+            ui.message("Created dataset at {}.".format(res['path']))
+        else:
             ui.message("Nothing was created")
```

Comment:
Snippet from `--help`:

````
  --output-format {simple,json,tailored}
                        select format for returned command results. 'simple'
                        give one line per result using a '<status>: <path>'
                        syntax; 'json' renders a JSON object with all
                        properties for each result (one per line); 'tailored'
                        enables a command-specific rendering style that is
                        typically tailored to human consumption (no result
                        output otherwise).
  --report-status {success,failure,ok,notneeded,impossible,error}
                        constrain command result report to records matching
                        the given status. 'success' is a synonym for 'ok' OR
                        'notneeded', 'failure' stands for 'impossible' OR
                        'error'.
  --report-type {dataset,file}
                        constrain command result report to records matching
                        the given type. Can be given more than once to match
                        multiple types.
  --on-failure {ignore,continue,stop}
                        when an operation fails: 'ignore' and continue with
                        remaining operations, the error is logged but does not
                        lead to a non-zero exit code of the command;
                        'continue' works like 'ignore', but an error causes a
                        non-zero exit code; 'stop' halts on first failure and
                        yields non-zero exit code. A failure is any result
                        with status 'impossible' or 'error'.
````